cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i

```diff
@@ -523,7 +593,7 @@ DAAL_FORCEINLINE void PredictClassificationTask<double, avx512>::predictByTree(c
         {
             checkMask = 0;
             size_t i  = 0;
-            for (size_t i = 0; i < _DEFAULT_BLOCK_SIZE; i += 8)
+            for (size_t i = 0; i < _blockSize; i += 8)
```

Comment:
It looks like there is a restriction: A `_blockSize` should actually be a multiple of 8. But the code does not checks this; and it can be easily broken if `_blockSize % 8 != 0`.
I think it would be better to have a hyperparameter defined as `_blockSizeMultiplier` and define the `_blockSize` as  `_blockSize = 8 * _blockSizeMultiplier`.
Then `_blockSizeMultiplier` won't have additional restrictions, it should only be positive and that's it.

Reply:
This assumption contradicts with the value `_DEFAULT_BLOCK_SIZE_COMMON = 22`. Previously, `_DEFAULT_BLOCK_SIZE` and `_DEFAULT_BLOCK_SIZE_COMMON` were independent. But now we're  setting both via the `blockSize` hyperparam. I strongly assume this is fine, because I don't see any reason why the `_DEFAULT_BLOCK_SIZE_COMMON` value would have to be `22`. So I will just always initialize to `32` and require `blockSize % blockSizeMultiplier == 0`. This will allow us to benchmark all scenarios later, even the `22` case.

Reply:
It actually does not contradict with `_DEFAULT_BLOCK_SIZE_COMMON = 22`.
Because there are 2 separate versions of `predictByTree` method:

1. `predictByTreeCommon`. This one can accept any value of `_blockSize`, even 22.
2. PredictClassificationTask<float[double], **avx512**>::predictByTree. This one can accept only `_blockSize` which is a multiple of 16 (8 for double), because it involves SIMD computations written with intrinsics. This part of the code just can't handle any value of `_blockSize`, unfortunately.

I see that `_DEFAULT_BLOCK_SIZE` and `_DEFAULT_BLOCK_SIZE_COMMON` are now merged into a single hyperparameter. I think it is a good decision to do so. But now some additional logic needs to be implemented in order not to break the code accidentally by passing incorrect (i.e. not divisible by 8[16]) value of `_blockSize` into  `PredictClassificationTask<float[double], avx512>::predictByTree`.